python/sglang/srt/layers/layernorm.py

```diff
@@ -0,0 +1,43 @@
+"""
+Copyright 2023-2024 SGLang Team
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+from typing import Optional, Tuple, Union
+
+import torch
+import torch.nn as nn
+
+
+class RMSNorm(nn.Module):
```

Comment:
I am not sure whether we should use CustomOp https://github.com/vllm-project/vllm/blob/825b044863a8e3af82a82a80cd2617486cc829ca/vllm/model_executor/layers/layernorm.py#L10

Reply:
I’ll take a look.

Reply:
It provides a base class, as well as abstractions for multiple backends.
https://github.com/vllm-project/vllm/blob/9fadc7b7a03f798036d0e8710587870e13bae759/vllm/model_executor/custom_op.py#L50-L62

Reply:
If we plan to add AMD support through Triton or FlashInfer in the future, then the current abstraction of CustomOp is not very useful for us. I'll close this for now and we can continue the discussion if there are any issues.